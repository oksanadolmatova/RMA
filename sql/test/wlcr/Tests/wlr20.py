from __future__ import print_function

try:
    from MonetDBtesting import process
except ImportError:
    import process
import os, sys, socket, time

dbfarm = os.getenv('GDK_DBFARM')
tstdb = os.getenv('TSTDB')

if not tstdb or not dbfarm:
    print('No TSTDB or GDK_DBFARM in environment')
    sys.exit(1)

def freeport():
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sock.bind(('', 0))
    port = sock.getsockname()[1]
    sock.close()
    return port

cloneport = freeport()

dbname = tstdb
dbnameclone = tstdb + '-clone'

# master = process.server(dbname = dbname, stdin = process.PIPE, stdout = process.PIPE, stderr = process.PIPE)
slave = process.server(dbname = dbnameclone, mapiport = cloneport, stdin = process.PIPE, stdout = process.PIPE, stderr = process.PIPE)

c = process.client('sql', dbname = dbnameclone, port = cloneport, stdin = process.PIPE, stdout = process.PIPE, stderr = process.PIPE)

#two step roll forward, where first step shouldn't do anything because already in previous test
cout, cerr = c.communicate('''\
call replicate('%s',1);
select * from tmp;
call replicate('%s',2);
select * from tmp;
call replicate('%s',4);
select * from tmp;
''' % (dbname,dbname,dbname))

sout, serr = slave.communicate()
#mout, merr = master.communicate()

#sys.stdout.write(mout)
sys.stdout.write(sout)
sys.stdout.write(cout)
#sys.stderr.write(merr)
sys.stderr.write(serr)
sys.stderr.write(cerr)
