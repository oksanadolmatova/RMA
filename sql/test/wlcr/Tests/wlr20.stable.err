stderr of test 'wlr20` in directory 'sql/test/wlcr` itself:


# 13:22:29 >  
# 13:22:29 >  "/usr/bin/python2" "wlr20.py" "wlr20"
# 13:22:29 >  

# builtin opt 	gdk_dbpath = /export/scratch1/mk/wlcr//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	monet_prompt = 
# cmdline opt 	mapi_port = 58415
# cmdline opt 	mapi_usock = /var/tmp/mtest-3357/.s.monetdb.58415
# cmdline opt 	gdk_dbpath = /export/scratch1/mk/wlcr//Linux/var/MonetDB/mTests_sql_test_wlcr-clone
# cmdline opt 	gdk_debug = 536870922

# 13:22:30 >  
# 13:22:30 >  "Done."
# 13:22:30 >  

