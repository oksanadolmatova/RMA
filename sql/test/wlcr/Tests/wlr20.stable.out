stdout of test 'wlr20` in directory 'sql/test/wlcr` itself:


# 13:22:29 >  
# 13:22:29 >  "/usr/bin/python2" "wlr20.py" "wlr20"
# 13:22:29 >  

# MonetDB 5 server v11.26.0
# This is an unreleased version
# Serving database 'mTests_sql_test_wlcr-clone', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2017 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:58415/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-3357/.s.monetdb.58415
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
#call replicate('mTests_sql_test_wlcr',1);
#Waiting for replay scheduler to stop
#replay log file:/export/scratch1/mk/wlcr//Linux/var/MonetDB/mTests_sql_test_wlcr/wlc_logs//mTests_sql_test_wlcr_000000000001
#select * from tmp;
% sys.tmp,	sys.tmp # table_name
% i,	s # name
% int,	clob # type
% 1,	0 # length
#call replicate('mTests_sql_test_wlcr',2);
#Waiting for replay scheduler to stop
#replay log file:/export/scratch1/mk/wlcr//Linux/var/MonetDB/mTests_sql_test_wlcr/wlc_logs//mTests_sql_test_wlcr_000000000001
#select * from tmp;
% sys.tmp,	sys.tmp # table_name
% i,	s # name
% int,	clob # type
% 1,	5 # length
[ 1,	"hello"	]
[ 2,	"world"	]
#call replicate('mTests_sql_test_wlcr',4);
#Waiting for replay scheduler to stop
#replay log file:/export/scratch1/mk/wlcr//Linux/var/MonetDB/mTests_sql_test_wlcr/wlc_logs//mTests_sql_test_wlcr_000000000002
#replay log file:/export/scratch1/mk/wlcr//Linux/var/MonetDB/mTests_sql_test_wlcr/wlc_logs//mTests_sql_test_wlcr_000000000003
#select * from tmp;
% sys.tmp,	sys.tmp # table_name
% i,	s # name
% int,	clob # type
% 1,	5 # length
[ 1,	"hello"	]
[ 2,	"world"	]
[ 3,	"blah"	]
[ 4,	"bloh"	]
[ 5,	"red"	]
[ 6,	"fox"	]

# 13:22:30 >  
# 13:22:30 >  "Done."
# 13:22:30 >  

